app/controllers/pensions/UpdatePensionProviderController.scala

```diff
@@ -97,15 +103,16 @@ trait UpdatePensionProviderController extends TaiBaseController
             UpdateRemovePensionForm.form.bindFromRequest().fold(
               formWithErrors => {
                 val model = PensionProviderViewModel(mandatoryVals.head.toInt, mandatoryVals.last)
-                Future(BadRequest(views.html.pensions.doYouGetThisPensionIncome(model, formWithErrors)))
+                Future(BadRequest(views.html.pensions.update.doYouGetThisPensionIncome(model, formWithErrors)))
               },
               {
-                case Some(YesValue) => Future.successful(
-                  Redirect(controllers.pensions.routes.UpdatePensionProviderController.whatDoYouWantToTellUs()))
+                case Some(YesValue) =>
+                  journeyCacheService.cache(UpdatePensionProvider_ReceivePensionQuestionKey, Messages("tai.label.yes")).map { _ =>
+                    Redirect(controllers.pensions.routes.UpdatePensionProviderController.whatDoYouWantToTellUs())
```

Comment:
do other journeys need to follow suit in terms of caching the massage key rather than form constant?